### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by nick-pestell**
This PR introduces improves unit tests by adding additional test commands for testing edge cases of filtering. It also adds a test method test the `build_multi_fasta()` function. 

In `build_snp_matrix.py` there are additional checks on input `**kwargs` for filtering the sample set which raise exceptions or warnings. There are also additional comments.

I think this is sufficient in terms of unit tests.

### CODE REVIEW

File: unit_tests.py [hunk 1]
```diff
@@ -20,67 +19,100 @@ def test_remove_duplicates(self):
     def test_get_indexes_to_remove(self):
         # define dataframe for input
         test_df = pd.DataFrame({"submission":pd.Series(["1", "2", "2", "2", "1", "3"], dtype="object"),
-                                "pcMapped":pd.Series([0.1, 0.2, 0.3, 0.4, 0.1, 0.6], dtype=float)})
+                                "pcMapped":pd.Series([0.1, 0.2, 0.3, 0.4, 0.2, 0.6], dtype=float)})
```
Comment (ellisrichardj) [inline unit_tests.py:22]: Just checking that 5th entry for this is meant to be `0.2`?  In lines 15 and 28 its `0.1`, but otherwise the dataframes are the same.

**Review by ellisrichardj — COMMENTED** [review]
All looking good - this is coming along nicely, just one small point for clarification

**Review by ellisrichardj — APPROVED** [review]
Great - thanks for confirming
